### ISSUE TITLE
rtmp pull

### ISSUE THREAD

**Issue by spinningsand** [roju/tiktok-live-recorder#2]
I had one user that it couldn't monitor and I got this error:

`ERROR - rtmp_pull_url not in response: {'data': {'prompts': ''}, 'extra': {'now': 1692122709147}, 'status_code': 4003110}`

**Comment by roju** [comment]
I've gotten this error before when using the TOR proxy. It seems to be some kind of IP blacklist/throttling by TikTok. 

@Marksmanship256 I have not encountered this particular case where the rtmp_pull_url key exists but the value is empty. Is anyone else getting responses like this in the get_live_url function?

**Comment by spinningsand** [comment]
Could it have something to do with a username change? I just had it happen again on someone I used to be able to record, but after they changed their username, I started getting this error.

**Comment by spinningsand** [comment]
Or maybe a username change could be 1 of multiple causes.

**Comment by spinningsand** [comment]
> > Or maybe a username change could be 1 of multiple causes.
> 
> I have had the issue with 2 uerss
> 
> accounts had not changed their username accounts were not private They have both worked for a long time prior I don't use a proxy or vpn

Same here. That was the case with me the first time it happened. I was just suggesting what may be a potential cause and there may be more than 1. Or looking into it might help to find a root cause.

**Comment by roberto-sartori-gl** [comment]
I have the same issue with certain accounts, not sure why but the json appears mostly empty. From a chrome browser, the same url provides the correct data.
Maybe some headers stuff?

**Comment by roberto-sartori-gl** [comment]
@roju Ok I did some testing, and I found out that (on a browser, Chrome) I can see the full json from the https://webcast.tiktok.com/webcast/room/info/?aid=1988&room_id=${room_id} page only when I'm logged.

For example, I cannot read it in incognito mode (I get the same result as I get with the live-recorder tool).

Maybe tiktok is starting to enforce the login to view lives?

**Comment by spinningsand** [comment]
> @roju Ok I did some testing, and I found out that (on a browser, Chrome) I can see the full json from the [https://webcast.tiktok.com/webcast/room/info/?aid=1988&room_id=${room_id}](https://webcast.tiktok.com/webcast/room/info/?aid=1988&room_id=$%7Broom_id%7D) page only when I'm logged.
> 
> For example, I cannot read it in incognito mode (I get the same result as I get with the live-recorder tool).
> 
> Maybe tiktok is starting to enforce the login to view lives?
> 
> Edit: I added a cookie section to the headers in the bot_utils.py file, using the cookie from my Chrome logged istanance and well, the recording started fine. Without it, I couldn't even start it.
> 
> Can someone confirm this? @spinningsand

@roberto-sartori-gl Yeah, I can test it out. Can you share the code for the cookie section you added?

**Comment by spinningsand** [comment]
Still having this issue. Are there any more leads on what it could be?
